packages/flame/lib/src/components/position_component.dart

```diff
@@ -1,218 +1,360 @@
+
+  /// The scale factor of this component. The scale can be different along
+  /// the X and Y dimensions. A scale greater than 1 makes the component
+  /// bigger, and less than 1 smaller. The scale can also be negative,
+  /// which results in a mirror reflection along the corresponding axis.
+  NotifyingVector2 get scale => _transform.scale;
+  set scale(Vector2 scale) => _transform.scale = scale;
+
+  /// Anchor point for this component. An anchor point describes a point
+  /// within the rectangle of size [size]. This point is considered to
+  /// be the logical "center" of the component. This can be visualized
+  /// as the point where Flame "grabs" the component. All transforms
+  /// occur around this point: the [position] is where the anchor point
+  /// will end up after the component is translated; the rotation and
+  /// scaling also happen around this anchor point.
+  Anchor _anchor;
+  Anchor get anchor => _anchor;
+  set anchor(Anchor anchor) {
+    _anchor = anchor;
+    _onModifiedSizeOrAnchor();
+  }
 
-  /// Get the relative top left position regardless of the anchor and angle
-  Vector2 get topLeftPosition {
-    return anchor.toOtherAnchorPosition(
-      position,
-      Anchor.topLeft,
-      scaledSize,
-    );
+  /// The logical size of the component. The game assumes that this is the
+  /// approximate size of the object that will be drawn on the screen.
+  /// This size will therefore be used for collision detection and tap
+  /// handling.
+  ///
+  /// This property can be reassigned at runtime, although this is not
+  /// recommended. Instead, in order to make the [PositionComponent] larger
+  /// or smaller, change its [scale].
+  final NotifyingVector2 _size;
+  Vector2 get size => _size;
+  set size(Vector2 size) => _size.setFrom(size);
+
+  /// The width of the component in local coordinates. Note that the object
+  /// may visually appear larger or smaller due to application of [scale].
+  double get width => _size.x;
+  set width(double w) => _size.x = w;
+
+  /// The height of the component in local coordinates. Note that the object
+  /// may visually appear larger or smaller due to application of [scale].
+  double get height => _size.y;
+  set height(double h) => _size.y = h;
+
+  /// The "physical" size of the component. This is the size of the
+  /// component as seen from the parent's perspective, and it is equal to
+  /// [size] * [scale]. This is a computed property and cannot be
+  /// modified by the user.
+  final Vector2 _scaledSize;
+  Vector2 get scaledSize => _scaledSize;
+
+  /// Measure the distance (in parent's coordinate space) between this
+  /// component's anchor and the [other] component's anchor.
+  double distance(PositionComponent other) =>
+      position.distanceTo(other.position);
+
+  //----------------------------------------------------------------------------
+  // Coordinate transformations
+  //----------------------------------------------------------------------------
```

Comment:
I am not sure if these kind of comments is something we want to add, and if we do why not use a code folding structure like the commonly used `//#region YourRegion` and `//#endregion` syntax

Reply:
Thanks, I didn't know about this syntax!